libs/sdk-model/src/execution/filter/filterMerge.ts

```diff
@@ -92,11 +77,33 @@ export function mergeFilters(originalFilters: IFilter[], addedFilters: IFilter[]
     const attributeFilters = [...original.attribute, ...added.attribute];
 
     // merge date filters by date dataset qualifier
-    // added date filters should win, so they are specified first, unionBy prefers items from the first argument
-    const dateFilters = unionBy(added.date, original.date, filterObjectRef);
+    const dateFilters = mergeDateFilters(original.date, added.date);
 
     // concat measure value filters
     const measureValueFilters = [...original.measureValue, ...added.measureValue];
 
     return [...attributeFilters, ...dateFilters, ...measureValueFilters];
 }
+
+function mergeDateFilters(originalFilters: IDateFilter[], addedFilters: IDateFilter[]): IDateFilter[] {
+    const allFilters = [...originalFilters, ...addedFilters];
+    const grouped = groupBy(allFilters, (f) => objRefToString(filterObjRef(f)));
+
+    return Object.values(grouped).reduce((filters: IDateFilter[], filtersForDimension) => {
+        // use the last filter for the dimension specified.
+        // this makes sure that the added filter wins if it is specified
+        const lastFilterForDimension = last(filtersForDimension)!;
+
+        // if the last filter is all time, clear filters for this dimension, otherwise use the last filter
+        const isAddedFilterAllTime =
+            isRelativeDateFilter(lastFilterForDimension) &&
+            (isNil(relativeDateFilterValues(lastFilterForDimension).from) ||
+                isNil(relativeDateFilterValues(lastFilterForDimension).to));
```

Comment:
is this the right way test whether filter is all time? i thought that all time filters had some weird 'fake' ALL_TIME granularity in v7. how is it on v8?

also, i suggest to add the 'all-time' filter testing function as a top-level model fun so that the concept is captured first-class.

Reply:
Added the support for all time filters as a first class concept.